core/internal/filetransfer/file_transfers.go

```diff
@@ -10,8 +10,8 @@ type FileTransfer interface {
 	Download(task *DefaultDownloadTask) error
 }
 
-type ReferenceArtifactFileTransfer interface {
-	Upload(task *ReferenceArtifactUploadTask) error
+type ArtifactFileTransfer interface {
+	Upload(task *DefaultUploadTask) error
```

Comment:
I've changed the name to better reflect the interface, since these file transfers are not solely dealing with reference artifacts